gossip/client_test.go

```diff
@@ -138,26 +138,32 @@ func TestClientGossip(t *testing.T) {
 		}
 	}()
 
-	if err := local.AddInfo("local-key", nil, time.Second); err != nil {
+	if err := local.AddInfo("local-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
-	if err := remote.AddInfo("remote-key", nil, time.Second); err != nil {
+	if err := remote.AddInfo("remote-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
 
 	// Use an insecure context. We're talking to tcp socket which are not in the certs.
 	rpcContext := rpc.NewContext(&base.Context{Insecure: true}, nil, stopper)
-	client.start(local, disconnected, rpcContext, stopper)
+	disconnected <- client
 
-	util.SucceedsSoon(t, func() error {
```

Comment:
Panics lose error information, but I haven't found the additional information in `SucceedsSoon` to aid tracking down these flakiness problems. I imagine if the tests were failing deterministically the error info could be useful. The exponential backoff isn't that much better than simply sleeping for 10s of milliseconds.

Not sure exactly what timeout you think should be raised globally. `base.NetworkTimeout`? I did try allowing the dial timeout to be configurable in `rpc.Context`, but for some reason that didn't help with flakiness. I did see that network connections were no longer timing out trying to connect.

I've switched back to using `util.SucceedsSoon` here to satisfy the peanut gallery. Could have sworn that was still flaky last night, but it seems to be running stably right now.